docs/hello_nextflow/06_hello_config.md

```diff
@@ -1,247 +1,503 @@
+[a6/0f72fd] GATK_JOINTGENOTYPING     [100%] 1 of 1 ✔
+```
+
+Feel free to try it out with the Docker profile too. You just have to switch `-profile conda_on` to `-profile docker_on` in the command.
 
 ### Takeaway
 
-You know how to modify process resources using configuration files.
+You know how to use profiles to select a preset configuration at runtime with minimal hassle.
 
 ### What's next?
 
-Learn how to change the executor used by Nextflow.
+Learn how to change the executor used by Nextflow to actually do the work.
 
 ---
 
-## 4. Executor
-
-### 4.1. Local executor
+## 3. Determine what executor(s) should be used to do the work
 
 Until now, we have been running our pipeline with the local executor.
 This runs each step on the same machine that Nextflow is running on.
-However, for large genomics pipelines, you will want to use a distributed executor.
+However, for large workloads, you will typically want to use a distributed executor such as an HPC or cloud.
 Nextflow supports several different distributed executors, including:
 
 -   HPC (SLURM, PBS, SGE)
 -   AWS Batch
 -   Google Batch
 -   Azure Batch
 -   Kubernetes
+-   GA4GH TES
 
-We can modify the executor used by nextflow using the `executor` process directive. Because `local` is the default executor, the following configuration is implied:
+The executor is subject to a process directive called `executor`. By default it is set to `local`, so the following configuration is implied:
 
-```groovy title="nextflow.config" linenums="18"
+```groovy title="Built-in configuration"
 process {
     executor = 'local'
 }
 ```
 
-### 4.2. Other executors
+Let's look at what it would take to using a Slurm scheduler, assuming we had a connection to a cluster and Slurm was installed appropriately.
 
-!!! note
+!!! warning
+
+    What follows is for demonstration purposes but **will not execute the work** since we don't have access to an external executor.
 
-    This is a demonstration and designed to go wrong!
+### 3.1. Set up a Slurm executor
 
-If we wish to change executor, we could simply set this to one of the values in the documentation:
+Add the following lines to the `nextflow.config` file:
 
-```groovy title="nextflow.config" linenums="18"
+```groovy title="nextflow.config" linenums="12"
 process {
     executor = 'slurm'
 }
 ```
 
-However, if we add this to our config and run the pipeline we will that includes this error:
+And... that's it! As noted before, this does assume that Slurm itself is already set up for you, but this is really all Nextflow itself needs to know.
+
+Basically we are telling Nextflow to generate a Slurm submission script and submit it using an `sbatch` command.
 
-```console
-Cannot run program "sbatch"
+### 3.2. Launch the workflow to generate the job submission script
+
+Let's try running this; even though we now it won't execute (since we don't have Slurm set up in the Gitpod environment) we'll be able to see what the submission script looks like.
```

Comment:
```suggestion
Let's try running this; even though we know it won't execute (since we don't have Slurm set up in the Gitpod environment) we'll be able to see what the submission script looks like.
```